test/functional/wallet_importmulti.py

```diff
@@ -458,6 +477,147 @@ def run_test (self):
                 "timestamp": "",
             }])
 
+        # Import P2WPKH address as watch only
+        self.log.info("Should import a P2WPKH address as watch only")
+        address = self.nodes[0].getaddressinfo(self.nodes[0].getnewaddress(address_type="bech32"))
+        result = self.nodes[1].importmulti([{
+            "scriptPubKey": {
+                "address": address['address']
+            },
+            "timestamp": "now",
+        }])
+        assert_equal(result[0]['success'], True)
+        address_assert = self.nodes[1].getaddressinfo(address['address'])
+        assert_equal(address_assert['iswatchonly'], True)
+        assert_equal(address_assert['solvable'], False)
```

Comment:
@sipa can you remind me why a Native Pay-to-Witness-Public-Key-Hash isn't `solvable`, even though legacy Pay-to-Public-Key-Hash addresses are? Is that just to reduce "evil magic" like in the P2SH-P2PKH case?
If so, maybe add a comment in the test or in the RPC doc?

Reply:
In this example it's not solvable because the public key isn't included. It has nothing to do with the script/witness hashing type.

In general, solvability is very easy. Start at the scriptPubKey. Whenever it includes a script hash, we must receive the actual script, and recurse into it. Whenever you encounter a pubkey hash, we must receive the actual public key. A scriptPubKey is solvable whenever you can descend all the way to scripts and pubkeys without any hashes of unknown things.

The only exception to that is that whenever you have a private key X, we also automatically import the P2WPKH and P2SH-P2WPKH versions of it, so no separate importing is needed in that case.

Reply:
> In this example it's not solvable because the public key isn't included.

OK, that makes sense. Actually I misread the [legacy test](https://github.com/Sjors/bitcoin/blob/02d553b3e6fa78aa76cd825135a496590a243595/test/functional/wallet_importmulti.py#L134-L149): it does add the public key, which why it's solvable.